core/dbt/parser/models.py

```diff
@@ -26,46 +27,109 @@ def render_update(
+                experimentally_parsed['sources'] = source_calls
+
+            except ExtractionError as e:
+                experimentally_parsed = e
+
+        # normal dbt run
+        if not flags.USE_EXPERIMENTAL_PARSER:
+            # normal rendering
+            super().render_update(node, config)
+            # if we're sampling, compare for correctness
+            if sample:
+                value = []
+                # experimental parser couldn't parse
+                if isinstance(experimentally_parsed, Exception):
+                    value += ["01_experimental_parser_couldn't_parse"]
+                else:
+                    # look for false positive configs
+                    for c in experimentally_parsed['configs']:
+                        if c not in config._config_calls:
+                            value += ["02_false_positive_config_value"]
+                            break
+
+                    # look for missed configs
+                    for c in config._config_calls:
+                        if c not in experimentally_parsed['configs']:
+                            value += ["03_missed_config_value"]
+                            break
+
+                    # look for false positive sources
+                    for c in experimentally_parsed['sources']:
+                        if c not in node.sources:
+                            value += ["04_false_positive_source_value"]
+                            break
+
+                    # look for missed sources
+                    for c in node.sources:
+                        if c not in experimentally_parsed['configs']:
+                            value += ["05_missed_source_value"]
+                            break
+
+                    # look for false positive refs
+                    for c in experimentally_parsed['refs']:
+                        if c not in node.refs:
+                            value += ["06_false_positive_ref_value"]
+                            break
 
-                config._config_calls = config_calls
+                    # look for missed refs
+                    for c in node.refs:
+                        if c not in experimentally_parsed['refs']:
+                            value += ["07_missed_ref_value"]
+                            break
 
-                # this uses the updated config to set all the right things in the node
-                # if there are hooks present, it WILL render jinja. Will need to change
-                # when we support hooks
-                self.update_parsed_node(node, config)
+                    # dedup values
+                    value = list(set(value))
 
-                # udpate the unrendered config with values from the file
-                # values from yaml files are in there already
-                node.unrendered_config.update(dict(res['configs']))
+                    # set sample results
+                    # TODO set this somewhere so it can be sent
```

Comment:
Right on, @gshank or @kwigley could help with setting up snowplow tracking for this.

We could swing this with just a standard structured event (no schema addition necessary), but I think we'll want a new validation schema that checks for the possible outcome values you've defined above.

Reply:
Yeah I have no idea what you might expect to see in the database, so these are just wet noodle values I flung at the wall. What values make sense to you?

Reply:
The ones you picked make a lot of sense to me! Just a few tweaks